container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+				log.Fatalf("Failed to format the string accordingly at %s: %v", value, err)
+			}
+		}
+	}
+	if len(labelsArray) > 0 {
+		labelsMap := make(map[string]string)
+		for k, v := range labels {
+			labelsMap[k] = v
+		}
+		configFile.Config.Labels = labelsMap
+	}
+
+	if len(ports) > 0 {
+		if len(configFile.Config.ExposedPorts) == 0 {
+			configFile.Config.ExposedPorts = make(map[string]struct{})
+		}
+		for _, port := range ports {
+			if strings.Contains(port, "/") {
+				// the port spec has the form 80/tcp, 1234/udp so simply use it as the key.
+				configFile.Config.ExposedPorts[port] = struct{}{}
+			} else {
+				// assume tcp
+				configFile.Config.ExposedPorts[port+"/tcp"] = struct{}{}
+			}
+		}
+	}
+
+	if len(volumes) > 0 {
+		if len(configFile.Config.Volumes) == 0 {
+			configFile.Config.Volumes = make(map[string]struct{})
+		}
+		for _, volume := range volumes {
+			configFile.Config.Volumes[volume] = struct{}{}
+		}
+	}
+
+	if *workdir != "" {
+		configFile.Config.WorkingDir = *workdir
+	}
+
+	layers := []string{}
+	for _, l := range layer {
+		newLayer, err := extractValue(l)
+		if err != nil {
+			log.Fatalf("Failed to extract the contents of layer file: %v", err)
+		}
+		layers = append(layers, newLayer)
+	}
+	// diffIDs are ordered from bottom-most to top-most
+	// []Hash type
+	diffIDs := configFile.RootFS.DiffIDs
+	if len(layer) > 0 {
+		var diffIDToAdd v1.Hash
+		for _, layer := range layers {
+			if layer != emptyFile {
+				diffIDToAdd = v1.Hash{Algorithm: "sha256", Hex: layer}
+				diffIDs = append(diffIDs, diffIDToAdd)
+			}
+		}
+		configFile.RootFS = v1.RootFS{Type: "layers", DiffIDs: diffIDs}
+	}
+
+	// length of history is expected to match the length of diff_ids
+	history := configFile.History
+	var historyToAdd v1.History
+	var currAuthor string
+	if configFile.Author != "" {
+		currAuthor = configFile.Author
+	} else {
+		currAuthor = "Unknown"
+	}
+	var currCreated v1.Time
+	var zeroedVal v1.Time
+	if configFile.Created == zeroedVal {
+		currCreated = configFile.Created
+	} else {
+		currCreated = v1.Time{time.Unix(0, 0)}
+	}
+
+	for _, l := range layers {
+		historyToAdd = v1.History{
+			Author:    currAuthor,
+			Created:   currCreated,
+			CreatedBy: "bazel build ...",
+		}
+		if l == emptyFile {
+			historyToAdd.EmptyLayer = true
+		}
+		history = append([]v1.History{historyToAdd}, history...)
+	}
+	configFile.History = history
+
+	if len(entrypointPrefix) != 0 {
+		newEntrypoint := append(configFile.Config.Entrypoint, entrypointPrefix...)
+		configFile.Config.Entrypoint = newEntrypoint
+	}
+
+	// write out the updated config
+	rawConfig, err := json.Marshal(configFile)
+	if err != nil {
+		log.Fatalf("Unable to read config struct into json object: %v", err)
+	}
+	err = ioutil.WriteFile(*outputConfig, rawConfig, os.ModePerm)
+	if err != nil {
+		log.Fatalf("Writing config to %s was unsuccessful: %v", *outputConfig, err)
+	}
+
+	log.Println("Successfully created Image Config.")
```

Comment:
nit: Move these messages to right after the call to write out the config and manifest respectively and add the name of the generated file paths in the messages.